endpoints/put_test.go

```diff
@@ -0,0 +1,1093 @@
+	reqBody := "{\"puts\":[{\"type\":\"xml\",\"value\":\"text longer than size limit\"}]}"
+
+	// Use mock client that will return an error
+	backend := newErrorReturningBackend()
+
+	// Run client
+	router := httprouter.New()
+	m := metricstest.CreateMockMetrics()
+	router.POST("/cache", NewPutHandler(backend, m, 10, true))
+
+	_, httpTestRecorder := doMockPut(t, router, reqBody)
+
+	// Assert status code
+	assert.Equal(t, http.StatusInternalServerError, httpTestRecorder.Code, "Put should have failed because we are using an MockReturnErrorBackend")
+	//   metrics
+	assert.Equal(t, int64(1), metricstest.MockCounters["puts.current_url.request.total"], "Handle function should record every incomming PUT request")
+	assert.Equal(t, int64(0), metricstest.MockCounters["puts.current_url.request.custom_key"], "Custom key was provided for put request and was not accounted for")
+	assert.Equal(t, int64(0), metricstest.MockCounters["puts.current_url.request.bad_request"], "Bad request wasn't recorded")
+	assert.Equal(t, int64(1), metricstest.MockCounters["puts.current_url.request.error"], "WriteGetResponse error should have been recorded")
+	assert.Equal(t, 0.00, metricstest.MockHistograms["puts.current_url.duration"], "Successful GET request should have recorded duration")
+}
+
+func TestEmptyPutRequests(t *testing.T) {
+	type aTest struct {
+		description      string
+		reqBody          string
+		expectedResponse string
+		emptyResponses   bool
+	}
+	testCases := []aTest{
+		{
+			description:      "Blank value in put element",
+			reqBody:          "{\"puts\":[{\"type\":\"xml\",\"value\":\"\"}]}",
+			expectedResponse: "{\"responses\":[\"uuid\":\"\"]}",
+			emptyResponses:   false,
+		},
+		// This test is meant to come right after the "Blank value in put element" test in order to assert the correction
+		// of a bug in the pre-PR#64 version of `endpoints/put.go`
+		{
+			description:      "All empty body. ",
+			reqBody:          "{}",
+			expectedResponse: "{\"responses\":[]}",
+			emptyResponses:   true,
+		},
+		{
+			description:      "Empty puts arrray",
+			reqBody:          "{\"puts\":[]}",
+			expectedResponse: "{\"responses\":[]}",
+			emptyResponses:   true,
+		},
+	}
+
+	for i, test := range testCases {
+		// Set up server
+		backend := backends.NewMemoryBackend()
+		m := metricstest.CreateMockMetrics()
+		router := httprouter.New()
+		router.POST("/cache", NewPutHandler(backend, m, 10, true))
+		rr := httptest.NewRecorder()
+
+		// Create request everytime
+		request, err := http.NewRequest("POST", "/cache", strings.NewReader(test.reqBody))
+		assert.NoError(t, err, "[%d] Failed to create a POST request: %v", i, err)
+
+		// Run
+		router.ServeHTTP(rr, request)
+		assert.Equal(t, http.StatusOK, rr.Code, "[%d] ServeHTTP(rr, request) failed = %v \n", i, rr.Result())
+
+		// validate results
+		if test.emptyResponses && !assert.Equal(t, test.expectedResponse, rr.Body.String(), "[%d] Text response not empty as expected", i) {
+			return
+		}
```

Comment:
Is this conditional necessary? I think it can be deleted since we check `test.emptyResponses` on line 668.

Reply:
It wasn't. Refactored test case and added an additional scenario.

Reply:
It was not necessary. Entire test has been refactored and an additional test scenario was added.